crates/map/src/finders/dfs.cairo

```diff
@@ -0,0 +1,184 @@
+    fn search(grid: felt252, width: u8, height: u8, from: u8, to: u8) -> Span<u8> {
+        // [Check] The start and target are walkable
+        if Bitmap::get(grid, from) == 0 || Bitmap::get(grid, to) == 0 {
+            return array![].span();
+        }
+
+        // [Effect] Initialize the start and target nodes
+        let start = NodeTrait::new(from, 0, 0, 0);
+        let target = NodeTrait::new(to, 0, 0, 0);
+
+        // [Effect] Initialize visited nodes and parents
+        let mut visited: Felt252Dict<bool> = Default::default();
+        let mut parents: Felt252Dict<u8> = Default::default();
+
+        // [Compute] Start the recursive DFS
+        let found = Self::dfs_recursive(
+            grid, width, height, start, target, ref visited, ref parents
+        );
+
+        // Reconstruct and return the path if found
+        if found {
+            Finder::path_with_parents(ref parents, start, target)
+        } else {
+            array![].span()
+        }
+    }
+
+    /// Recursive helper function for DFS
+    #[inline]
+    fn dfs_recursive(
+        grid: felt252,
+        width: u8,
+        height: u8,
+        current: Node,
+        target: Node,
+        ref visited: Felt252Dict<bool>,
+        ref parents: Felt252Dict<u8>
+    ) -> bool {
+        // [Check] Mark current node as visited
+        visited.insert(current.position.into(), true);
+
+        // [Check] If we've reached the target, we're done
+        if current.position == target.position {
+            return true;
+        }
+
+        // [Compute] Evaluate the neighbors for all 4 directions
+        let seed = Seeder::shuffle(grid, current.position.into());
+        let mut directions = DirectionTrait::compute_shuffled_directions(seed);
+        let mut found = false;
+
+        loop {
+            if directions == 0 {
+                break;
+            }
+            let direction = DirectionTrait::pop_front(ref directions);
+            if Finder::check(grid, width, height, current.position, direction, ref visited) {
+                let neighbor_position = direction.next(current.position, width);
+                let neighbor = NodeTrait::new(neighbor_position, current.position, 0, 0);
+
+                // [Effect] Set parent for the neighbor
+                parents.insert(neighbor_position.into(), current.position);
+
+                // [Recurse] Continue DFS from the neighbor
+                found =
+                    Self::dfs_recursive(
+                        grid, width, height, neighbor, target, ref visited, ref parents
+                    );
+
+                if found {
+                    break;
+                }
+            }
+        };
+
+        // [Check] Return whether we've found the target
+        found
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::DFS;
+
+    #[test]
+    fn test_dfs_search_small() {
+        // x───┐   111    8 7 6
+        // 1 0 │   101    5 4 3
+        // 0 1 s   011    2 1 0
```

Comment:
You can remove extra comments

Reply:
Added them so that someone might get confused , so i thought it will improve readability